boot/freeldr/freeldr/ntldr/arch/i386/winldr.c

```diff
@@ -744,3 +744,45 @@ MempDump(VOID)
     }
 }
 #endif
+
+#ifdef UEFIBOOT
+PLOADER_PARAMETER_BLOCK PubLoaderBlockVA;
+KERNEL_ENTRY_POINT PubKiSystemStartup;
+
+extern ULONG LoaderPagesSpanned;
+
+VOID _LeaveUEFIStack(VOID);
+
+VOID
+WinldrFinalizeBoot(PLOADER_PARAMETER_BLOCK LoaderBlockVA,
+                   KERNEL_ENTRY_POINT KiSystemStartup)
+{
+    TRACE("Preparing to jump to kernel\n");
+    PubLoaderBlockVA = LoaderBlockVA;
+    PubKiSystemStartup = KiSystemStartup;
+    _LeaveUEFIStack();
+}
+
+VOID
+UefiExitToKernel(VOID)
```

Comment:
Hmm.... this looks redundant and duplicates what's already done in winldr.c .
It's not possible to return execution back to the caller? Is it because you've changed the stack?

Reply:
It’s not possible to return from callee because I’ve changed the stack. But yes I’m trying to work on figuring out a reliable way where I don’t have to change the stack, if that’s possible. I feel like it should be